gqlalchemy/query_builder.py

```diff
@@ -373,20 +376,55 @@ def construct_query(self) -> str:
 
 
 class ReturnPartialQuery(PartialQuery):
-    def __init__(self, results: Dict[str, str]):
+    def __init__(
+        self, results: Optional[Union[str, Tuple[str, str], List[Union[str, Tuple[str, str]]], Dict[str, str]]] = None
+    ):
         super().__init__(DeclarativeBaseTypes.RETURN)
 
-        self._results = results
-
-    @property
-    def results(self) -> str:
-        return self._results if self._results is not None else ""
+        if results is None:
+            self.query = None
+        elif isinstance(results, list):
+            self.query = self._return_read_list(results)
+        elif isinstance(results, dict):
+            self.query = self._return_read_dict(results)
+        else:
+            self.query = self._return_read_item(results)
 
     def construct_query(self) -> str:
         """Creates a RETURN statement Cypher partial query."""
-        if len(self.results) == 0:
+        if self.query is None:
             return " RETURN * "
-        return f" RETURN {dict_to_alias_statement(self.results)} "
+
+        return f" {self.type} {self.query} "
+
+    def _return_read_item(self, item: Union[str, Tuple[str, str]]) -> str:
+        if isinstance(item, str):
+            return f"{self._return_read_str(item)}"
+        elif isinstance(item, tuple):
+            return f"{self._return_read_tuple(item)}"
+        else:
+            raise GQLAlchemyReturnTypeError
+
+    def _return_read_list(self, results: List[Union[str, Tuple[str, str]]]):
+        return ", ".join(self._return_read_item(item=item) for item in results)
+
+    def _return_read_str(self, item: str) -> str:
```

Comment:
Why... This... ?

Reply:
Expand your comment please

Reply:
We discussed this, this method is not necessary and it was used in the `OrderByPartialQuery`. Now it has been removed from there and `ReturnPartialQuery`.